tests/test_avro_schema.py

```diff
@@ -2,16 +2,270 @@
+
+    msg = "evolving a field type to a union is a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema6, schema1)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "removing a type from a union is NOT a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema1, schema6)
+    assert res != SchemaCompatibilityResult.compatible(), msg
+
+    msg = "adding a new type in union is a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema7, schema6)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "removing a type from a union is NOT a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema6, schema7)
+    assert res != SchemaCompatibilityResult.compatible(), msg
+
+
+def test_schemaregistry_basic_backwards_transitive_compatibility():
+    """
+    Backward transitive compatibility: A new schema is backward compatible if it can be used to read the data
+    written in all previous schemas.
+    """
+    msg = "iteratively adding fields with defaults is a compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema8, schema1)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema8, schema2)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "adding a field with default is a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema2, schema1)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "removing a default is a compatible change, but not transitively"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema3, schema2)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "removing a default is not a transitively compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema3, schema2)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema3, schema1)
+    assert res != SchemaCompatibilityResult.compatible(), msg
+
+
+def test_schemaregistry_basic_forwards_compatibility():
+    """
+    Forward compatibility: A new schema is forward compatible if the previous schema can read data written in this
+    schema.
+    """
+    msg = "adding a field is a forward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema1, schema2)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "adding a field is a forward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema1, schema3)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "adding a field is a forward compatible change"
```

Comment:
Does this and the next one really check adding a field or it's more about defaults?